build fails with clang 22 due to Werror

`clang version 22.1.1`

```
at 00:12:26 ❯ make
make  all-recursive
make[1]: Entering directory '/tmp/FreedroidClassic'
Making all in src
make[2]: Entering directory '/tmp/FreedroidClassic/src'
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W -Werror -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-BFont.o -MD -MP -MF .deps/freedroid-BFont.Tpo -c -o freedroid-BFont.o `test -f 'BFont.c' || echo './'`BFont.c
BFont.c:328:13: error: assigning to 'char *' from 'const char *' discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
  328 |   while ((p = strchr (&text[pos + 1], ' ')) != NULL)
      |             ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BFont.c:378:10: error: assigning to 'char *' from 'const char *' discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
  378 |               p = strstr (&text[pos + 1], " ");
      |                 ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BFont.c:645:8: error: cast from 'Uint8 *' (aka 'unsigned char *') to 'Uint16 *' (aka 'unsigned short *') increases required alignment from 1 to 2 [-Werror,-Wcast-align]
  645 |       *(Uint16 *) p = pixel;
      |        ^~~~~~~~~~~~
BFont.c:664:8: error: cast from 'Uint8 *' (aka 'unsigned char *') to 'Uint32 *' (aka 'unsigned int *') increases required alignment from 1 to 4 [-Werror,-Wcast-align]
  664 |       *(Uint32 *) p = pixel;
      |        ^~~~~~~~~~~~
4 errors generated.
make[2]: *** [Makefile:555: freedroid-BFont.o] Error 1
make[2]: Leaving directory '/tmp/FreedroidClassic/src'
make[1]: *** [Makefile:465: all-recursive] Error 1
make[1]: Leaving directory '/tmp/FreedroidClassic'
make: *** [Makefile:363: all] Error 2
```

after manually removing `-Werror`
```
make[2]: Entering directory '/tmp/FreedroidClassic/src'
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-BFont.o -MD -MP -MF .deps/freedroid-BFont.Tpo -c -o freedroid-BFont.o `test -f 'BFont.c' || echo './'`BFont.c
BFont.c:328:13: warning: assigning to 'char *' from 'const char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  328 |   while ((p = strchr (&text[pos + 1], ' ')) != NULL)
      |             ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BFont.c:378:10: warning: assigning to 'char *' from 'const char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  378 |               p = strstr (&text[pos + 1], " ");
      |                 ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BFont.c:645:8: warning: cast from 'Uint8 *' (aka 'unsigned char *') to 'Uint16 *' (aka 'unsigned short *') increases required alignment from 1 to 2 [-Wcast-align]
  645 |       *(Uint16 *) p = pixel;
      |        ^~~~~~~~~~~~
BFont.c:664:8: warning: cast from 'Uint8 *' (aka 'unsigned char *') to 'Uint32 *' (aka 'unsigned int *') increases required alignment from 1 to 4 [-Wcast-align]
  664 |       *(Uint32 *) p = pixel;
      |        ^~~~~~~~~~~~
4 warnings generated.
mv -f .deps/freedroid-BFont.Tpo .deps/freedroid-BFont.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-bullet.o -MD -MP -MF .deps/freedroid-bullet.Tpo -c -o freedroid-bullet.o `test -f 'bullet.c' || echo './'`bullet.c
bullet.c:281:14: warning: variable 'FBTZaehler' set but not used [-Wunused-but-set-variable]
  281 |   static int FBTZaehler = 0;
      |              ^
1 warning generated.
mv -f .deps/freedroid-bullet.Tpo .deps/freedroid-bullet.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-enemy.o -MD -MP -MF .deps/freedroid-enemy.Tpo -c -o freedroid-enemy.o `test -f 'enemy.c' || echo './'`enemy.c
mv -f .deps/freedroid-enemy.Tpo .deps/freedroid-enemy.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-graphics.o -MD -MP -MF .deps/freedroid-graphics.Tpo -c -o freedroid-graphics.o `test -f 'graphics.c' || echo './'`graphics.c
graphics.c:1057:15: warning: cast from 'Uint8 *' (aka 'unsigned char *') to 'Uint16 *' (aka 'unsigned short *') increases required alignment from 1 to 2 [-Wcast-align]
 1057 |       return *(Uint16 *)p;
      |               ^~~~~~~~~~~
graphics.c:1066:15: warning: cast from 'Uint8 *' (aka 'unsigned char *') to 'Uint32 *' (aka 'unsigned int *') increases required alignment from 1 to 4 [-Wcast-align]
 1066 |       return *(Uint32 *)p;
      |               ^~~~~~~~~~~
graphics.c:1095:10: warning: cast from 'Uint8 *' (aka 'unsigned char *') to 'Uint16 *' (aka 'unsigned short *') increases required alignment from 1 to 2 [-Wcast-align]
 1095 |         *(Uint16 *)p = pixel;
      |          ^~~~~~~~~~~
graphics.c:1111:10: warning: cast from 'Uint8 *' (aka 'unsigned char *') to 'Uint32 *' (aka 'unsigned int *') increases required alignment from 1 to 4 [-Wcast-align]
 1111 |         *(Uint32 *)p = pixel;
      |          ^~~~~~~~~~~
4 warnings generated.
mv -f .deps/freedroid-graphics.Tpo .deps/freedroid-graphics.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-highscore.o -MD -MP -MF .deps/freedroid-highscore.Tpo -c -o freedroid-highscore.o `test -f 'highscore.c' || echo './'`highscore.c
mv -f .deps/freedroid-highscore.Tpo .deps/freedroid-highscore.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-influ.o -MD -MP -MF .deps/freedroid-influ.Tpo -c -o freedroid-influ.o `test -f 'influ.c' || echo './'`influ.c
mv -f .deps/freedroid-influ.Tpo .deps/freedroid-influ.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-init.o -MD -MP -MF .deps/freedroid-init.Tpo -c -o freedroid-init.o `test -f 'init.c' || echo './'`init.c
init.c:996:31: warning: assigning to 'char *' from 'const char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  996 |       if ( (TerminationPointer=strstr ( NextSubsectionStartPointer, END_OF_BRIEFING_SUBSECTION_STRING)) == NULL)
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
mv -f .deps/freedroid-init.Tpo .deps/freedroid-init.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-input.o -MD -MP -MF .deps/freedroid-input.Tpo -c -o freedroid-input.o `test -f 'input.c' || echo './'`input.c
mv -f .deps/freedroid-input.Tpo .deps/freedroid-input.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-level_editor.o -MD -MP -MF .deps/freedroid-level_editor.Tpo -c -o freedroid-level_editor.o `test -f 'level_editor.c' || echo './'`level_editor.c
mv -f .deps/freedroid-level_editor.Tpo .deps/freedroid-level_editor.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-main.o -MD -MP -MF .deps/freedroid-main.Tpo -c -o freedroid-main.o `test -f 'main.c' || echo './'`main.c
mv -f .deps/freedroid-main.Tpo .deps/freedroid-main.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-map.o -MD -MP -MF .deps/freedroid-map.Tpo -c -o freedroid-map.o `test -f 'map.c' || echo './'`map.c
mv -f .deps/freedroid-map.Tpo .deps/freedroid-map.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-menu.o -MD -MP -MF .deps/freedroid-menu.Tpo -c -o freedroid-menu.o `test -f 'menu.c' || echo './'`menu.c
mv -f .deps/freedroid-menu.Tpo .deps/freedroid-menu.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-misc.o -MD -MP -MF .deps/freedroid-misc.Tpo -c -o freedroid-misc.o `test -f 'misc.c' || echo './'`misc.c
misc.c:348:23: warning: assigning to 'char *' from 'const char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  348 |   if ( (SearchPointer = strstr ( SearchString , StartIndicationString )) == NULL )
      |                       ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
misc.c:578:15: warning: assigning to 'char *' from 'const char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  578 |   if ( ( temp = strstr ( SearchBeginPointer , SearchTextPointer ) ) == NULL)
      |               ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 warnings generated.
mv -f .deps/freedroid-misc.Tpo .deps/freedroid-misc.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-sdl_compat.o -MD -MP -MF .deps/freedroid-sdl_compat.Tpo -c -o freedroid-sdl_compat.o `test -f 'sdl_compat.c' || echo './'`sdl_compat.c
mv -f .deps/freedroid-sdl_compat.Tpo .deps/freedroid-sdl_compat.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-ship.o -MD -MP -MF .deps/freedroid-ship.Tpo -c -o freedroid-ship.o `test -f 'ship.c' || echo './'`ship.c
mv -f .deps/freedroid-ship.Tpo .deps/freedroid-ship.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-sound.o -MD -MP -MF .deps/freedroid-sound.Tpo -c -o freedroid-sound.o `test -f 'sound.c' || echo './'`sound.c
mv -f .deps/freedroid-sound.Tpo .deps/freedroid-sound.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-takeover.o -MD -MP -MF .deps/freedroid-takeover.Tpo -c -o freedroid-takeover.o `test -f 'takeover.c' || echo './'`takeover.c
mv -f .deps/freedroid-takeover.Tpo .deps/freedroid-takeover.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-text.o -MD -MP -MF .deps/freedroid-text.Tpo -c -o freedroid-text.o `test -f 'text.c' || echo './'`text.c
text.c:196:7: warning: variable 'Number_Of_Line_Feeds' set but not used [-Wunused-but-set-variable]
  196 |   int Number_Of_Line_Feeds = 0;         /* Anzahl der Textzeilen */
      |       ^
1 warning generated.
mv -f .deps/freedroid-text.Tpo .deps/freedroid-text.Po
/home/matthias/.cargo/bin/sccache clang -DHAVE_CONFIG_H -I. -I..  -DLOCAL_DATADIR=\"/tmp/FreedroidClassic\"  -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"' -MT freedroid-view.o -MD -MP -MF .deps/freedroid-view.Tpo -c -o freedroid-view.o `test -f 'view.c' || echo './'`view.c
mv -f .deps/freedroid-view.Tpo .deps/freedroid-view.Po
/home/matthias/.cargo/bin/sccache clang -Wall -W  -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation -g -O2  -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/libpng16 -DHWY_SHARED_DEFINE -DAVIF_DLL -I/usr/include/rav1e -I/usr/include/svt-av1 -DEB_DLL -DRTC_BUILD=0 -I/usr/include/webp -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT  -DFD_DATADIR='"/usr/local/share/freedroid"'   -o freedroid freedroid-BFont.o freedroid-bullet.o freedroid-enemy.o freedroid-graphics.o freedroid-highscore.o freedroid-influ.o freedroid-init.o freedroid-input.o freedroid-level_editor.o freedroid-main.o freedroid-map.o freedroid-menu.o freedroid-misc.o freedroid-sdl_compat.o freedroid-ship.o freedroid-sound.o freedroid-takeover.o freedroid-text.o freedroid-view.o   -lvorbis -lpng -lz -ljpeg -lm  -lSDL2 -lSDL2_image -lSDL2 -lSDL2_mixer -lSDL2
make[2]: Leaving directory '/tmp/FreedroidClassic/src'
Making all in map
make[2]: Entering directory '/tmp/FreedroidClassic/map'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/tmp/FreedroidClassic/map'
Making all in graphics
make[2]: Entering directory '/tmp/FreedroidClassic/graphics'
Making all in classic_theme
make[3]: Entering directory '/tmp/FreedroidClassic/graphics/classic_theme'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/tmp/FreedroidClassic/graphics/classic_theme'
Making all in lanzz_theme
make[3]: Entering directory '/tmp/FreedroidClassic/graphics/lanzz_theme'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/tmp/FreedroidClassic/graphics/lanzz_theme'
Making all in para90_theme
make[3]: Entering directory '/tmp/FreedroidClassic/graphics/para90_theme'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/tmp/FreedroidClassic/graphics/para90_theme'
make[3]: Entering directory '/tmp/FreedroidClassic/graphics'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/tmp/FreedroidClassic/graphics'
make[2]: Leaving directory '/tmp/FreedroidClassic/graphics'
Making all in sound
make[2]: Entering directory '/tmp/FreedroidClassic/sound'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/tmp/FreedroidClassic/sound'
make[2]: Entering directory '/tmp/FreedroidClassic'
make[2]: Leaving directory '/tmp/FreedroidClassic'
make[1]: Leaving directory '/tmp/FreedroidClassic'
```

--- a/.github/workflows/c-cpp.yml
+++ b/.github/workflows/c-cpp.yml
@@ -7,8 +7,23 @@ on:
     branches: [ master ]
 
 jobs:
-  build:
-    runs-on: ubuntu-latest
+  ubuntu-build:
+    name: ubuntu/${{ matrix.name }}
+    runs-on: ${{ matrix.runs_on }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: gcc
+            runs_on: ubuntu-24.04
+            cc: gcc
+            make_target: all
+          - name: clang-strict
+            runs_on: ubuntu-latest
+            cc: clang
+            make_target: strict
+    env:
+      CC: ${{ matrix.cc }}
 
     steps:
     - uses: actions/checkout@v4
@@ -32,4 +47,51 @@ jobs:
       run: ./configure
 
     - name: Build
-      run: make -j"$(nproc)"
+      run: make -j"$(nproc)" ${{ matrix.make_target }}
+
+  arch-build:
+    name: arch/${{ matrix.name }}
+    runs-on: ubuntu-latest
+    container:
+      image: archlinux:latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: gcc
+            cc: gcc
+            make_target: all
+          - name: clang-strict
+            cc: clang
+            make_target: strict
+    env:
+      CC: ${{ matrix.cc }}
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Install build dependencies
+      run: |
+        pacman -Syu --noconfirm --needed \
+          autoconf \
+          automake \
+          gcc \
+          clang \
+          libjpeg-turbo \
+          libpng \
+          libtool \
+          libvorbis \
+          make \
+          pkgconf \
+          sdl2 \
+          sdl2_image \
+          sdl2_mixer
+
+    - name: Bootstrap autotools
+      run: ./00boot
+
+    - name: Configure
+      run: ./configure
+
+    - name: Build
+      run: make -j"$(nproc)" ${{ matrix.make_target }}

--- a/Makefile.am
+++ b/Makefile.am
@@ -4,6 +4,11 @@ man_MANS = freedroid.6
 
 EXTRA_DIST = $(man_MANS) mac-osx
 
+.PHONY: strict
+
+strict:
+	$(MAKE) WERROR_CFLAGS=-Werror all
+
 ## Win32 specific stuff follows here
 
 win_txts = COPYING AUTHORS README Releasetext freedroid-man
@@ -34,4 +39,3 @@ android-data:
 	mkdir -p AndroidData
 	zip -r AndroidData/data-1.2.2.zip graphics/ sound/*.wav sound/*.ogg map/
 	cp graphics/title.jpg AndroidData/logo.jpg
-

--- a/src/BFont.c
+++ b/src/BFont.c
@@ -321,7 +321,7 @@ TextWidthFont (BFont_Info * Font, const char *text)
 int
 count (const char *text)
 {
-  char *p = NULL;
+  const char *p = NULL;
   int pos = -1;
   int i = 0;
   /* Calculate the space occupied by the text without spaces */
@@ -349,7 +349,7 @@ JustifiedPutStringFont (SDL_Surface * Surface, BFont_Info * Font, int y,
   int dif;
 
   char *strtmp;
-  char *p;
+  const char *p;
   int pos = -1;
   int xpos = 0;
 
@@ -642,7 +642,10 @@ PutPixel (SDL_Surface * surface, int x, int y, Uint32 pixel)
       break;
 
     case 2:
-      *(Uint16 *) p = pixel;
+      {
+        Uint16 pixel16 = (Uint16) pixel;
+        memcpy (p, &pixel16, sizeof (pixel16));
+      }
       break;
 
     case 3:
@@ -661,7 +664,7 @@ PutPixel (SDL_Surface * surface, int x, int y, Uint32 pixel)
       break;
 
     case 4:
-      *(Uint32 *) p = pixel;
+      memcpy (p, &pixel, sizeof (pixel));
       break;
     }
 }

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-AM_CFLAGS = -Wall -W -Werror -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation
+AM_CFLAGS = -Wall -W $(WERROR_CFLAGS) -Wunused -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common -Wnested-externs -Wno-format-zero-length -fno-strict-aliasing -Wno-unused-result -Wno-unknown-pragmas -Wno-misleading-indentation
 
 bin_PROGRAMS = freedroid
 
@@ -58,3 +58,8 @@ freedroidDebug_CPPFLAGS = $(freedroid_CPPFLAGS)
 WINDRES = i586-mingw32msvc-windres
 freedroid.coff: freedroid.rc ../graphics/paraicon.ico
 	$(WINDRES) -i freedroid.rc -o freedroid.coff -O COFF
+
+.PHONY: strict
+
+strict:
+	$(MAKE) WERROR_CFLAGS=-Werror all

--- a/src/bullet.c
+++ b/src/bullet.c
@@ -278,7 +278,6 @@ CheckBulletCollisions (int num)
   int levelnum = CurLevel->levelnum;
   float xdist, ydist;
   Bullet CurBullet = &AllBullets[num];
-  static int FBTZaehler = 0;
   finepoint step;
   int num_check_steps, stepnum;
   int i;
@@ -408,7 +407,6 @@ CheckBulletCollisions (int num)
 
 		  if (!CurBullet->mine)
 		    {
-		      FBTZaehler++;
 		    }
 		  CurBullet->type = OUT;
 		  CurBullet->mine = FALSE;

--- a/src/graphics.c
+++ b/src/graphics.c
@@ -1054,7 +1054,11 @@ getpixel(SDL_Surface *surface, int x, int y)
       return *p;
 
     case 2:
-      return *(Uint16 *)p;
+      {
+        Uint16 pixel16;
+        memcpy (&pixel16, p, sizeof (pixel16));
+        return pixel16;
+      }
 
     case 3:
       if(SDL_BYTEORDER == SDL_BIG_ENDIAN)
@@ -1063,7 +1067,11 @@ getpixel(SDL_Surface *surface, int x, int y)
 	return p[0] | p[1] << 8 | p[2] << 16;
 
     case 4:
-      return *(Uint32 *)p;
+      {
+        Uint32 pixel32;
+        memcpy (&pixel32, p, sizeof (pixel32));
+        return pixel32;
+      }
 
     default:
       return 0;       /* shouldn't happen, but avoids warnings */
@@ -1092,7 +1100,10 @@ void putpixel(SDL_Surface *surface, int x, int y, Uint32 pixel)
         break;
 
     case 2:
-        *(Uint16 *)p = pixel;
+        {
+          Uint16 pixel16 = (Uint16) pixel;
+          memcpy (p, &pixel16, sizeof (pixel16));
+        }
         break;
 
     case 3:
@@ -1108,7 +1119,7 @@ void putpixel(SDL_Surface *surface, int x, int y, Uint32 pixel)
         break;
 
     case 4:
-        *(Uint32 *)p = pixel;
+        memcpy (p, &pixel, sizeof (pixel));
         break;
     }
 } // void putpixel(...)

--- a/src/init.c
+++ b/src/init.c
@@ -237,7 +237,7 @@ void
 Get_Robot_Data ( void* DataPointer )
 {
   int RobotIndex = 0;
-  char *RobotPointer;
+  const char *RobotPointer;
 
   float maxspeed_calibrator;
   float acceleration_calibrator;
@@ -590,7 +590,7 @@ InitNewMission ( const char *MissionName )
   char *fpath;
   int i;
   char *MainMissionPointer;
-  char *BriefingSectionPointer;
+  const char *BriefingSectionPointer;
   char *StartPointPointer;
   char Buffer[500];
   int NumberOfStartPoints=0;
@@ -959,7 +959,7 @@ Title ( const char *MissionBriefingPointer )
 {
   const char* NextSubsectionStartPointer;
   char* PreparedBriefingText = NULL;
-  char* TerminationPointer;
+  const char* TerminationPointer;
   char Buffer[500];
   int ThisTextLength;
   SDL_Rect rect;

--- a/src/map.c
+++ b/src/map.c
@@ -967,7 +967,7 @@ GetLiftConnections (char *filename)
 {
   char *fpath;
   char *Data;
-  char *EntryPointer;
+  const char *EntryPointer;
   int i;
   int Label;
   int DeckIndex;
@@ -1170,7 +1170,7 @@ GetThisLevelsDroids( const char* SectionPointer )
 {
   int OurLevelNumber;
   const char* SearchPointer;
-  char* EndOfThisLevelData;
+  const char* EndOfThisLevelData;
   int MaxRand;
   int MinRand;
   int RealNumberOfRandomDroids;
@@ -1189,7 +1189,6 @@ GetThisLevelsDroids( const char* SectionPointer )
   // printf("\nReceived another levels droid section for decoding. It reads: %s " , SectionPointer );
 
   EndOfThisLevelData = LocateStringInData ( SectionPointer , DROIDS_LEVEL_END_INDICATION_STRING );
-  EndOfThisLevelData[0]=0;
 
   // Now we read in the level number for this level
   ReadValueFromString (SectionPointer, DROIDS_LEVEL_INDICATION_STRING, "%d", &OurLevelNumber);
@@ -1202,7 +1201,9 @@ GetThisLevelsDroids( const char* SectionPointer )
 
   DifferentRandomTypes=0;
   SearchPointer = SectionPointer;
-  while ( ( SearchPointer = strstr ( SearchPointer , ALLOWED_TYPE_INDICATION_STRING)) != NULL)
+  while ( SearchPointer < EndOfThisLevelData &&
+	  ( SearchPointer = strstr ( SearchPointer , ALLOWED_TYPE_INDICATION_STRING)) != NULL &&
+	  SearchPointer < EndOfThisLevelData)
     {
       SearchPointer += strlen ( ALLOWED_TYPE_INDICATION_STRING );
       strncpy( TypeIndicationString , SearchPointer , 3 ); // Every type is 3 characters long

--- a/src/misc.c
+++ b/src/misc.c
@@ -340,8 +340,8 @@ The original source string specified should in no way be modified.
 char*
 ReadAndMallocStringFromData ( const char* SearchString , const char* StartIndicationString , const char* EndIndicationString )
 {
-  char* SearchPointer;
-  char* EndOfStringPointer;
+  const char* SearchPointer;
+  const char* EndOfStringPointer;
   char* ReturnString = NULL;
   int StringLength;
 
@@ -570,10 +570,10 @@ The return value is a pointer to the first instance where the substring
 we are searching is found in the main text.
 ----------------------------------------------------------------------
 */
-char*
+const char*
 LocateStringInData ( const char* SearchBeginPointer, const char* SearchTextPointer )
 {
-  char* temp;
+  const char* temp;
 
   if ( ( temp = strstr ( SearchBeginPointer , SearchTextPointer ) ) == NULL)
     {
@@ -619,7 +619,7 @@ not resolve.... Sorry, if that interrupts a major game of yours.....\n\
 void
 ReadValueFromString (const char* data, const char* label, const char* FormatString, void* dst)
 {
-  char *pos;
+  const char *pos;
 
   // Now we locate the label in data and position pointer right after the label
   pos = LocateStringInData (data, label); // ..will Terminate itself if not found...

--- a/src/proto.h
+++ b/src/proto.h
@@ -272,7 +272,7 @@ EXTERN char* ReadAndMallocStringFromData ( const char* SearchString , const char
 EXTERN int CountStringOccurences ( const char* SearchString , const char* TargetString ) ;
 EXTERN void ReadValueFromString(const char* data, const char* label, const char* FormatString, void* dst);
 EXTERN char* ReadAndMallocAndTerminateFile( const char* filename , const char* File_End_String ) ;
-EXTERN char* LocateStringInData ( const char* SearchBeginPointer, const char* SearchTextPointer ) ;
+EXTERN const char* LocateStringInData ( const char* SearchBeginPointer, const char* SearchTextPointer ) ;
 EXTERN char* find_file (const char *fname, const char *subdir, int use_theme, int critical);
 EXTERN void CheckForTriggeredEvents ( void );
 EXTERN void Pause (void);

--- a/src/text.c
+++ b/src/text.c
@@ -193,8 +193,6 @@ AddInfluBurntText( void )
 int
 ScrollText (char *Text, SDL_Rect *rect)
 {
-  int Number_Of_Line_Feeds = 0;		/* Anzahl der Textzeilen */
-  char *textpt;			/* bewegl. Textpointer */
   float InsertLine = 1.0*rect->y;
   int speed = 30;   // in pixel / sec
   int maxspeed = 150;
@@ -206,12 +204,6 @@ ScrollText (char *Text, SDL_Rect *rect)
   Background = SDL_ConvertSurface (ne_screen, ne_screen->format, 0);
 
   // first_tick = SDL_GetTicks ();
-
-  // count the number of lines in the text
-  textpt = Text;
-  while (*textpt++)
-    if (*textpt == '\n')
-      Number_Of_Line_Feeds++;
 
   wait_for_all_keys_released();
   while (1)

